dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java

```diff
@@ -113,21 +114,23 @@ private String getExpression( String expression, DimensionalItemId id, List<Stri
 
     private List<String> getCategoryOptionCombosIntersection( List<String> cogUidList, String dataElementId )
     {
-        List<String> cocUidIntersection = null;
+        List<String> cocUidIntersection = new ArrayList<>();
 
         for ( String cogUid : cogUidList )
         {
-            CategoryOptionGroup cog = categoryOptionGroupStore
-                .getByUid( cogUid );
-
+            CategoryOptionGroup cog = categoryOptionGroupStore.getByUid( cogUid );
+            if ( cog == null )
+            {
+                continue;
```

Comment:
I think we should probably not silently ignore this and instead fail early, by using `loadByUid`, and throw an exception. I think we should encapsulate these null checks a bit more.

Reply:
I was also thinking about adding a `loadById(long)` that would throw an exception so we can use lookup by long and have the exception behaviour. WDYT?

Reply:
Yes I think that sounds good.

Reply:
Yes, meant in general - there were other cases I think.